lib/resty/router/router.lua

```
@@ -97,5 +97,38 @@ function _M:get_fields()
     return out
 end
 
+do 
+    local routers = {}
+    local DEFAULT_UUID = "00000000-0000-0000-0000-000000000000"
+    local DEFAULT_PRIORITY = 0
+
+    -- validate an expression against a schema
+    -- @param expr the expression to validate
+    -- @param schema the schema to validate against
+    -- @return true if the expression is valid, (nil, error) otherwise
+    function _M.validate(schema, expr)
+        local r = routers[schema]
+
+        if not r then
+            r = ffi_gc(clib.router_new(schema.schema), router_free)
+            routers[schema] = r
+        end
+
+        local errbuf = get_string_buf(ERR_BUF_MAX_LEN)
+        local errbuf_len = get_size_ptr()
+        errbuf_len[0] = ERR_BUF_MAX_LEN
+    
+        if clib.router_add_matcher(r, DEFAULT_PRIORITY, DEFAULT_UUID, expr,
+                                   errbuf, errbuf_len) == false then
+            return nil, ffi_string(errbuf, errbuf_len[0])
+        end
+
+        if clib.router_remove_matcher(r, DEFAULT_PRIORITY, DEFAULT_UUID) == false then
```

This should never happen, suggest a `assert` instead of hiding the error.